src/mempool/mempool.rs

```diff
@@ -354,7 +419,7 @@ impl Mempool {
                     // We want to pull in multiple messages per poll so that throughput is not blocked on the polling frequency. The number of messages we pull should be fixed and relatively small so that the mempool isn't always stuck here.
                     for _ in 0..256 {
                         if self.config.allow_unlimited_mempool_size || (self.messages.len() as u64) < self.config.capacity_per_shard {
-                            match self.mempool_rx.try_recv() {
+                            match self.read_node_mempool.mempool_rx.try_recv() {
```

Comment:
Hmm, this is looking too deeply into the struct. I know I said composition, but I think look at this, maybe a trait maybe a better way to share the code here. e.g. a `Mempool` trait that implements the validation/gossip etc. 

But this is not a blocking, we can refactor later.